### PR TITLE
 Adding structures for iot custom authorizer request/response 

### PR DESCRIPTION
**Pull request by ynori7**
Adding structures to fit to the documentation here: https://docs.aws.amazon.com/iot/latest/apireference/API_TestInvokeAuthorizer.html

### CODE REVIEW

**Comment by LegNeato** [comment]
@bmoffatt any chance we can get this merged?

File: events/iot.go
Comment (bmoffatt) [inline events/iot.go]: I don't see a context field in the documentation. Is there another reference that contains this?

from https://docs.aws.amazon.com/iot/latest/apireference/API_TestInvokeAuthorizer.html#API_TestInvokeAuthorizer_ResponseElements
```
{
   "disconnectAfterInSeconds": number,
   "isAuthenticated": boolean,
   "policyDocuments": [ "string" ],
   "principalId": "string",
   "refreshAfterInSeconds": number
}
```

File: events/iot.go
Comment (bmoffatt) [inline events/iot.go]: minor, It may make sense to add or use an existing IAMPolicy type instead of redefining the envelope. 

A README would also help to show expected usage.

**Comment by ynori7** [comment]
> @bmoffatt any chance we can get this merged?

Yeah, it looks like the contract changed a lot since I made these changes. I've updated everything now and addressed all the comments.

**Comment by ynori7** [comment]
@bmoffatt Could you take another look?

File: .idea/aws-lambda-go.iml
Comment (bmoffatt) [inline .idea/aws-lambda-go.iml:1]: project files need to be removed :)

File: events/iot.go
Comment (bmoffatt) [inline events/iot.go:18]: style: general field and  type naming conventions should consistently capitalize these initialisms 

```
Id -> ID
HTTP -> HTTP
Tls->TLS
Mqtt -> MQTT
```

**Review by bmoffatt — CHANGES_REQUESTED** [review]

File: events/policy.go
Comment (bmoffatt) [inline events/policy.go:1]: unused file?

Comment (bmoffatt) [reply events/policy.go:1]: going to delete this one before merging, feel free to re-open if this one was needed for something!

**Review by bmoffatt — APPROVED** [review]

File: events/policy.go
Comment (ynori7) [reply events/policy.go]: I think it was used before, but not anymore after resolving the most recent merge conflicts since it was added already somewhere else with a different name. Thanks for cleaning up!

**Comment by rittneje** [comment]
@ynori7 @bmoffatt Unfortunately, I think you referenced the wrong request structure. The actual structure is documented here: https://docs.aws.amazon.com/iot/latest/developerguide/config-custom-auth.html I will file a bug.

I guess for some reason the IoT Core developers decided to make the test request totally different from the real request.

**Pull request merged**
